plugins/flytekit-spark/flytekitplugins/spark/task.py

```diff
@@ -176,6 +184,27 @@ def get_custom(self, settings: SerializationSettings) -> Dict[str, Any]:
 
         return MessageToDict(job.to_flyte_idl())
 
+    def to_k8s_pod(self, pod_template: Optional[PodTemplate] = None) -> Optional[K8sPod]:
```

Comment:
We can use this [function](https://github.com/flyteorg/flytekit/blob/3d842a89bcecefec46e3f35fe272935fa770a687/flytekit/models/task.py#L1078) instead?